src/net/java/sip/communicator/impl/protocol/jabber/ChatRoomMemberJabberImpl.java

```diff
@@ -87,6 +89,14 @@
      */
     private String displayName;
 
+    /**
+     * A map to store {@link Object}s. Along with the provided methods
+     * {@link this#setValue(String, Object)} and {@link this#getValue(String)},
+     * this map prevents adding a getter/setter for each value which we want to
+     * associate with this {@link ChatRoomMember}.
+     */
+    private final Map<String, Object> valueMap = new HashMap<>();
```

Comment:
What is supposed to be stored here?

Reply:
The avatar-url, avatar-id or gravatar hash based on whatever is present, as well as group id and user-id from Stride organisation when deployed in that environment. Because there are so many different extensions used, it seemed better to add something like this instead of using a setter/getter for everything

Reply:
I'm not a fan of such maps but I can see the reasoning behind. @damencho ?

Reply:
I also see the reason behind that map. But we already have such mechanism for calls, @nikvaessen you can follow the same pattern. Just make ChatRoomMemberJabberImpl extend DataObject, where you will have getData(Object key) and setData(Object key, Object value). 

The other thing here is the newly introduced interface which exposes smack object in its interface, something we were trying to avoid. There is already one such precedent in OperationSetJitsiMeetTools... but, if we can avoid adding more ...

So you can add ChatRoomMemberPresenceChangeEvent.MEMBER_UPDATED and reuse ChatRoomMemberPresenceListener for your needs (as Boris proposed). And you can store lastReceivedPresence and add a method getLastPresence for ChatRoomMemberJabberImpl and handle storing of data that way, avoiding exposing smack API in the jitsi protocol providers API. 
Does this sound ok?